### PR TITLE
Complete Platform Validator PR

### PR DESCRIPTION
**Pull request by isabelmsft**
Original platform validator PR https://github.com/sonic-net/sonic-utilities/pull/2791/checks?check_run_id=12708751588 was merged and reverted due to unexpected test case behavior blocking submodule advance. The test case was fixed through https://github.com/sonic-net/sonic-mgmt/pull/8469. Now, this platform validator PR brings back the original changes

### CODE REVIEW

**Review by qiluo-msft — APPROVED** [review]

File: generic_config_updater/field_operation_validators.py [hunk 1]
```diff
@@ -1,10 +1,119 @@
...
+            gcu_field_operation_conf = json.load(s)
+    else:
+        raise GenericConfigUpdaterError("GCU table modification validators config file not found")
+    
+    asic_mapping = gcu_field_operation_conf["helper_data"]["rdma_config_update_validator"]
+    
+    if device_info.get_sonic_version_info()['asic_type'] == 'cisco-8000':
+        asic = "cisco-8000"
+    elif device_info.get_sonic_version_info()['asic_type'] == 'vs':
+        asic = "td2"
```
Comment (qiluo-msft) [inline generic_config_updater/field_operation_validators.py:27]: >asic [](http://example.com/codeflow?start=8&length=4)

This is not always true. Please check [https://github.com/sonic-net/sonic-buildimage/blob/220ea74cbb7e5cc783a27d7ca9f966899d81b7b3/platform/vs/docker-sonic-vs/Dockerfile.j2#L176](https://github.com/sonic-net/sonic-buildimage/blob/220ea74cbb7e5cc783a27d7ca9f966899d81b7b3/platform/vs/docker-sonic-vs/Dockerfile.j2#L176)

Comment (isabelmsft) [reply generic_config_updater/field_operation_validators.py]: Updated the PR to fully consider vs hwskus

File: generic_config_updater/field_operation_validators.py [hunk 2]
```diff
@@ -1,10 +1,116 @@
...
+        output, err = proc.communicate()
+        hwsku = output.rstrip('\n')
+        if asic_type == 'mellanox' or asic_type == 'vs':
+            spc1_hwskus = asic_mapping["mellanox_asics"]["spc1"]
+            if hwsku.lower() in [spc1_hwsku.lower() for spc1_hwsku in spc1_hwskus]:
+                asic = "spc1"
+        if asic_type == 'broadcom' or asic_type == 'vs':
+            broadcom_asics = asic_mapping["broadcom_asics"]
+            for asic_shorthand, hwskus in broadcom_asics.items():
+                if asic != "unknown":
```
Comment (qiluo-msft) [inline generic_config_updater/field_operation_validators.py:39]: >asic [](http://example.com/codeflow?start=19&length=4)

Should you check unknown much earlier?

Comment (isabelmsft) [reply generic_config_updater/field_operation_validators.py:39]: asic is set to "unknown" at the start of the function. It remains unknown until one of the specified asic types in the table is identified. The purpose of this particular check is to stop iterating over all possible broadcom asics once the asic is already identified.

Comment (qiluo-msft) [reply generic_config_updater/field_operation_validators.py:39]: If it is already identified as spc1, you can return immediately.

Comment (isabelmsft) [reply generic_config_updater/field_operation_validators.py:40]: Changed to return spc1 immediately

**Review by qiluo-msft — APPROVED** [review]

**Comment by isabelmsft** [comment]
I will open a new PR for 202205 branch, as this is not a clean cherry-pick

**Pull request merged**